src/images/kafka.rs

```diff
@@ -0,0 +1,124 @@
+use crate::{core::WaitFor, Image};
+use std::{collections::HashMap, path::PathBuf};
+
+const CONTAINER_IDENTIFIER: &str = "confluentinc/cp-kafka";
+const DEFAULT_TAG: &str = "6.1.1";
+
+pub const KAFKA_PORT: u16 = 9093;
+const ZOOKEEPER_PORT: u16 = 2181;
+
+#[derive(Clone, Debug)]
+pub struct KafkaArgs {
+    pub bash_pathname: PathBuf,
+    pub options: Vec<String>,
+    pub commands: String,
+}
+
+impl Default for KafkaArgs {
+    fn default() -> Self {
+        let starter_commands = format!(
+            "echo 'clientPort={}' > zookeeper.properties; \
+echo 'dataDir=/var/lib/zookeeper/data' >> zookeeper.properties; \
+echo 'dataLogDir=/var/lib/zookeeper/log' >> zookeeper.properties; \
+zookeeper-server-start zookeeper.properties & \
+export KAFKA_ZOOKEEPER_CONNECT='localhost:{}'; \
+export KAFKA_ADVERTISED_LISTENERS='PLAINTEXT://localhost:{},BROKER://localhost:9092'; \
+. /etc/confluent/docker/bash-config && \
+/etc/confluent/docker/configure && \
+/etc/confluent/docker/launch",
+            ZOOKEEPER_PORT, ZOOKEEPER_PORT, KAFKA_PORT
+        );
+
+        Self {
+            bash_pathname: PathBuf::from("/bin/bash"),
+            options: vec!["-c".to_owned()],
+            commands: starter_commands,
+        }
+    }
+}
+
+impl IntoIterator for KafkaArgs {
+    type Item = String;
+    type IntoIter = ::std::vec::IntoIter<String>;
+
+    fn into_iter(self) -> Self::IntoIter {
+        let mut args = vec![self.bash_pathname.into_os_string().into_string().unwrap()];
+        args.extend(self.options.into_iter());
+        args.push(self.commands);
+        args.into_iter()
+    }
+}
+
+#[derive(Debug)]
+pub struct Kafka {
+    arguments: KafkaArgs,
+    env_vars: HashMap<String, String>,
+    tag: String,
+}
+
+impl Kafka {
+    pub fn with_env_vars(mut self, env_vars: HashMap<String, String>) -> Self {
+        self.env_vars.extend(env_vars.into_iter());
+        self
+    }
+
+    pub fn with_tag<T: Into<String>>(self, tag: T) -> Self {
+        Self {
+            tag: tag.into(),
+            ..self
+        }
+    }
+}
+
+impl Default for Kafka {
+    fn default() -> Self {
+        let mut env_vars = HashMap::new();
+
+        env_vars.insert(
+            "KAFKA_LISTENER_SECURITY_PROTOCOL_MAP".to_owned(),
+            "BROKER:PLAINTEXT,PLAINTEXT:PLAINTEXT".to_owned(),
+        );
+        env_vars.insert("KAFKA_BROKER_ID".to_owned(), "1".to_owned());
+        env_vars.insert(
+            "KAFKA_OFFSETS_TOPIC_REPLICATION_FACTOR".to_owned(),
+            "1".to_owned(),
+        );
```

Comment:
These look like they are always static so you can use [`phf`](https://docs.rs/phf/0.8.0/phf/) to store them in a private const and then just reference that const in the `fn env_vars()` function.

Reply:
That'd be a great idea, except that there seems to be an impedance mismatch between the return type of `env_vars` and `phf::Map`? Let's say we have the following static map:
```
type KafkaEnvVarsMap = phf::Map<&'static str, &'static str>;

static KAFKA_ENV_VARS: KafkaEnvVarsMap = phf_map! {
    "KAFKA_ZOOKEEPER_CONNECT" => "localhost:2181",
    "KAFKA_ADVERTISED_LISTENERS" => "PLAINTEXT://localhost:9093,BROKER://localhost:9092",
    "KAFKA_LISTENER_SECURITY_PROTOCOL_MAP" => "BROKER:PLAINTEXT,PLAINTEXT:PLAINTEXT",
    "KAFKA_BROKER_ID" => "1",
    "KAFKA_OFFSETS_TOPIC_REPLICATION_FACTOR" => "1",
};
```
The body of `env_vars` returning a reference to the map does not look like it can satisfy the trait bound for `Image::EnvVars`: `Default + IntoIterator<Item = (String, String)>`, plus `phf::Map` does not seem to support owned Strings as a key type. Do we want to use `lazy_static` or `once_cell::sync::Lazy` instead if `pfh::Map` does not work?

Reply:
Bummer! Sorry to send you onto this goose-chase. I should have checked better before making this suggestion.

Let's just leave it as is then :)

Reply:
No problem, thanks for the suggestion! While on this topic, I removed `with_env_vars` as per [this](https://github.com/testcontainers/testcontainers-rs/pull/282#discussion_r612096632) conversation but started feeling that the user not being able to configure Kafka settings through environment variables can be a bit too restrictive. Maybe we can leave `with_env_vars` removed for now and keep an eye on users' request in the future?